internal/framework/types/smithy_json_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package types_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/attr/xattr"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	smithyjson "github.com/hashicorp/terraform-provider-aws/internal/json"
+)
+
+func TestSmithyJSONTypeValueFromTerraform(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		val      tftypes.Value
+		expected attr.Value
+	}{
+		"null value": {
+			val:      tftypes.NewValue(tftypes.String, nil),
+			expected: fwtypes.SmithyJSONNull[smithyjson.JSONStringer](),
+		},
+		"unknown value": {
+			val:      tftypes.NewValue(tftypes.String, tftypes.UnknownValue),
+			expected: fwtypes.SmithyJSONUnknown[smithyjson.JSONStringer](),
+		},
+		"valid SmithyJSON": {
+			val:      tftypes.NewValue(tftypes.String, `{"test": "value"}`),                      // lintignore:AWSAT003,AWSAT005
```

Comment:
I think the `lintignore` comments can be removed from all of these.

```suggestion
			val:      tftypes.NewValue(tftypes.String, `{"test": "value"}`),
```